editor/editor_node.cpp

```diff
@@ -2816,8 +2816,9 @@ Dictionary EditorNode::_get_main_scene_state() {
 	return state;
 }
 
-void EditorNode::_set_main_scene_state(Dictionary p_state, Node *p_for_scene) {
-
+void EditorNode::_set_main_scene_state(Dictionary p_state) {
+	//We fetch the current edited scene now, since we come out of a deferred call
+	Node *p_for_scene = get_edited_scene();
```

Comment:
It's weird that this is enough to fix the crash. What this does is basically to make the next line a tautology (`get_edited_scene() != p_for_scene` will always be false, since you just set them to be the same thing).

And then `p_for_scene` is not used anywhere else.

So it seems the crash happens when `_set_main_scene_state` does return early?

Reply:
No, the crash happens when Godot tries to validate the arguments for a deferred call, it doesn't even arrive into the _set_main_scene_state method. It fails in validating args(2) (crashes with both type of casts). After inspecting a bit, seems like the Node becomes invalid and when Godot tries to validate it it reads garbage memory (some bools have negative values and the whole node content seems pretty messed up)

Reply:
Ah that makes sense.

>  After inspecting a bit, seems like the Node becomes invalid and when Godot tries to validate it it reads garbage memory (some bools have negative values and the whole node content seems pretty messed up)

That sounds like a normal Variant::NIL with a null pointer. The other members of Variant are then uninitialized and contain garbage data indeed.